components/post-title/index.tsx

```diff
@@ -1,11 +1,18 @@
 import { useEntityProp } from '@wordpress/core-data';
 import { RichText, store as blockEditorStore } from '@wordpress/block-editor';
 import { useSelect } from '@wordpress/data';
-import PropTypes from 'prop-types';
 import { usePost } from '../../hooks';
 
-export const PostTitle = (props) => {
-	const { tagName: TagName = 'h1', ...rest } = props;
+interface PostTitleProps<T extends React.ElementType> {
+	tagName?: T;
+}
+
+type PostTitlePropsWithOmit<T extends React.ElementType> = PostTitleProps<T> & Omit<React.ComponentPropsWithoutRef<T>, keyof PostTitleProps<T>>;
+
+export const PostTitle = <T extends React.ElementType = 'h1'>({
+	tagName,
+	...rest
+}: PostTitlePropsWithOmit<T> ) => {
```

Comment:
Not at all critical, and I am very open to being wrong here. My personal preference over single letter type generics is having a more descriptive name. 
```suggestion
interface PostTitleProps<T extends React.ElementType> {
	tagName?: T;
}

type PostTitlePropsWithOmit<TElementType extends React.ElementType> = PostTitleProps<TElementType> & Omit<React.ComponentPropsWithoutRef<TElementType>, keyof PostTitleProps<TElementType>>;

export const PostTitle = <TElementType extends React.ElementType = 'h1'>({
	tagName,
	...rest
}: PostTitlePropsWithOmit<TElementType> ) => {
```